cmd/refinery/main.go

```diff
@@ -226,5 +227,8 @@ func main() {
 
 	// block on our signal handler to exit
 	sig := <-sigsToExit
+	// unregister ourselves before we go
+	close(done)
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Do we sleep here to give time for the unregistering to occur?  Is it possible that this amount of time would ever not be enough?  Would be great if we would block until we are 100% sure we've unregistered and then continue on.

Reply:
Yes, that's why the sleep. There's a way to make it safe, but it's really complicated to make it perfectly correct (channels of channels with complicated deadlock prevention) -- and the difference here is between other nodes seeing that you shut down in N seconds if it worked, and 3N seconds if it didn't work. I've actually implemented this before and I don't think the complexity is worth it.